### DISCUSSION TITLE
Bluetooth pressure sensor

### DISCUSSION THREAD

**Discussion by vlszabolcs** [OpenTracksApp/OpenTracks#1423] [Ideas]
I would like to use an external Bluetooth pressure sensor because my phone has't built in. I use Arduino Nano 33 BLE Sense dev board now. I not the java and android app expert so i need some help how can i implement this in OpenTracks if it's possible. 
Any tips or idea how can i start?
Thanks!

**Comment by dennisguse** [comment]
Are you building an air pressure sensor (i.e., barometer) to measure elevation gain/ loss?
If so great - my phone also lacks this sensor and I would love to get this running.

How to proceed.
1. Create a new issues for this (to track things)
2. Figure out the actual BLE protocol (service, characteristic, and announcement)
3. Implement the BLE protocol correctly
4. Implement discovery (settings) and parsing in OpenTracks
5. Actually use data in OpenTracks

Most relevant BLE code is in: `BluetoothUtils.java`

PS/ could it be that the best suiting BLE service is the environmental sensing service aka 181A?  https://www.bluetooth.com/specifications/specs/environmental-sensing-service-1-0/
providing the characteristic 2A6D: https://github.com/oesmith/gatt-xml/blob/master/org.bluetooth.characteristic.pressure.xml

Some intel from here: https://forum.arduino.cc/t/nano-33-sense-temp-humid-pressure-via-bluetooth-gives-wildly-incorrect-values/675933/3
and from all over the Internet.
For each new sensor, I am confused again.
